### PR TITLE
Adds docs for resolvers

### LINKED ISSUE #2080: Documentation on how to use ~/.m2 for dependencies
**Version(s)**
Latest

**Describe what needs to be done and why**
Its not very clear how i can use a dependency which i published to the maven local repository.

Quite often we developed libraries and publish them local maven repository. 

**Is your feature request related to a past ticket or discussion?**
Please mention any other issues or discussions relevant to this one.

**Describe alternatives you've considered**
To `COURSIER_REPOSITORIES=`  i tried adding 
`Dangerous.mavenLocal`
`mavenLocal`
`~/.m2`
`file://~/.m2`
But it seems coursier doesn't document the usage of a the maven local repository. 

**Additional context**
We only use maven/gradle to do the publishing so just chaning to an local ivy isn't trivial.

### PR DESCRIPTION
**Pull request by romanowski**
Fixes #2080

Includes a small clean up.

### CODE REVIEW

**Review by lwronski — COMMENTED** [review]
Add tests for parsing repository parameters.

File: website/docs/guides/dependencies.md [hunk 1]
```diff
@@ -37,6 +37,39 @@ For example:
 org.postgresql:postgresql:42.2.8
 ```
 
+### Repositories
+
+Sometimes dependencies are published into non-standard repositories, like nightly builds published to Sonatype Snapshots. Scala CLI can use additional maven and ivy repositories with  `repository` directive or `--reposiroty` command line options:
```
Comment (Gedochao) [inline website/docs/guides/dependencies.md]:
```suggestion
Sometimes dependencies are published into non-standard repositories, like nightly builds published to Sonatype Snapshots. Scala CLI can use additional maven and ivy repositories with the `repository` directive or `--repository` command line options:
```

File: website/docs/guides/dependencies.md [hunk 2]
```diff
@@ -37,6 +37,39 @@ For example:
...
+
+```scala
+//> using repository sonatype:snapshots
+```
+
+or
+
+```scala-cli --repository "https://maven-central.storage-download.googleapis.com/maven2"```
```
Comment (Gedochao) [inline website/docs/guides/dependencies.md]:
````suggestion
```bash ignore
scala-cli --repository "https://maven-central.storage-download.googleapis.com/maven2"
```
````

File: website/docs/guides/dependencies.md [hunk 3]
```diff
@@ -37,6 +37,39 @@ For example:
...
+
+Both directive and command line option accepts predefined repository definition (see below) or an URL to the root of Maven repository.
```
Comment (Gedochao) [inline website/docs/guides/dependencies.md]:
```suggestion
Both directive and command line option accept predefined repository definitions (see below) or a URL of the root of the Maven repository.
```

File: website/docs/guides/dependencies.md [hunk 4]
```diff
@@ -37,6 +37,39 @@ For example:
...
+
+Repositories may be also resolved from `COURSIER_REPOSITORIES` environment variable but this is not recommended (more in [Coursier documentation](https://get-coursier.io/docs/other-repositories)).
```
Comment (Gedochao) [inline website/docs/guides/dependencies.md]:
```suggestion
Repositories can also be resolved from the `COURSIER_REPOSITORIES` environment variable, but this is not recommended (more in [Coursier documentation](https://get-coursier.io/docs/other-repositories)).
```

File: website/docs/guides/dependencies.md [hunk 5]
```diff
@@ -37,6 +37,39 @@ For example:
...
+
+
+| predefined repository | kind | description  |
+| ---------- | ---- | --- |
+| central | Maven [(root)](https://repo1.maven.org/maven2) | Used by default, default repository for most Scala libraries |
+| sonatype:snapshots | Maven [(root)](https://oss.sonatype.org/content/repositories/snapshots) | Repositories where most Scala libraries publish its snapshots / nightly builds. Used when `X.nightly` is used as Scala version e.g. `3.1.nightly`. |
+| ivy2local | Ivy | Local ivy repository, used to publish things locally (e.g. by `publishLocal`). Localized in `<ivy-home>/local`, usually `<user-home>/.ivy/local`.    |
+| m2Local | Maven | Local maven repository, localized in `<user-home>/.m2/repository` |
+
+Scala CLI delegate parsing of predefined repositories to Coursier and full details can be obtains from Coursier source code ([here](https://github.com/coursier/coursier/blob/2444eebcc151e0f6927e269137e8737c1f31cbe2/modules/coursier/jvm/src/main/scala/coursier/LocalRepositories.scala) and [here](https://github.com/coursier/coursier/blob/2444eebcc151e0f6927e269137e8737c1f31cbe2/modules/coursier/shared/src/main/scala/coursier/internal/SharedRepositoryParser.scala))
```
Comment (Gedochao) [inline website/docs/guides/dependencies.md]:
```suggestion
Scala CLI delegates parsing of predefined repositories to Coursier and full details can be obtains from Coursier source code ([here](https://github.com/coursier/coursier/blob/2444eebcc151e0f6927e269137e8737c1f31cbe2/modules/coursier/jvm/src/main/scala/coursier/LocalRepositories.scala) and [here](https://github.com/coursier/coursier/blob/2444eebcc151e0f6927e269137e8737c1f31cbe2/modules/coursier/shared/src/main/scala/coursier/internal/SharedRepositoryParser.scala))
```

File: modules/directives/src/main/scala/scala/build/preprocessing/directives/Repository.scala [hunk 6]
```diff
@@ -36,4 +37,9 @@ final case class Repository(
 
 object Repository {
   val handler: DirectiveHandler[Repository] = DirectiveHandler.derive
+
+  val usageMsg =
+    """Add repositories for dependency resolution.
+      |
+      |Accepts predefined repository supported by Courier (like `sonatype:snapshots` or `m2Local`) or an URL to the root of Maven repository""".stripMargin
```
Comment (Gedochao) [inline modules/directives/src/main/scala/scala/build/preprocessing/directives/Repository.scala]:
```suggestion
      |Accepts predefined repositories supported by Coursier (like `sonatype:snapshots` or `m2Local`) or a URL of the root of Maven repository""".stripMargin
```

**Review by lwronski — APPROVED** [review]

**Pull request merged**